### PR TITLE
Upgrade to SUSHI 3.1.0 and GoFSH 2.0.1

### PR DESCRIPTION
**Pull request by jafeltra**
TLDR: Updated to SUSHI 3.3.2 and GoFSH 2.1.0. There is one warning resolved by including `exprContextCritical = false` in the `config-overrides.js` file. The tests require a resolver to correctly use antlr4 within the Jest testing environment.

The rest of the text below is kept for posterity. It should reflect the current state, but it is quite a journey. Proceed if you dare.

==========================================================================================

This PR updates the SUSHI version to ~v3.0.0~ ~v3.1.0~ ~v3.2.0~ v3.3.2. ~The only issue I had when updating SUSHI to ~3.0.0~ 3.1.0 was a few warnings in the console coming from the `tsconfig-paths` dependency that was introduced in that SUSHI version to handle ANTLR properly in SUSHI. I was able to null load two of the modules from the dependency without any negative effects in FSH Online. There is one remaining warning in the console coming from that dependency. When I null loaded that module, the app crashed, so something in it must be used. However, I don't see any negative consequences of that warning. I can't find a way to resolve the warning and keep the app working, so if anyone has other ideas about how to resolve it or why it is okay to ignore it, I'm very open to them.~

~_**Update on the above paragraph**: After updating to v3.2.0, SUSHI no longer uses the `tsconfig-paths` and everything in the grammar seems to be working as expected. However, the tests are not yet configured correctly with this new SUSHI version._~

_**Update on the update above**: The latest commits and SUSHI versions no longer use `tsconfig-paths` **and** properly configure the tests to work with the latest version of antlr4 used in the current SUSHI version._ 

This PR is also aiming to update GoFSH to v2.0.0. ~but GoFSH needs to be updated before it can be incorporated. However, as part of this PR review, you should try out using [this GoFSH PR](https://github.com/FHIR/GoFSH/pull/225) with this FSH Online branch. The issue with using the published 2.0.0 version of GoFSH seems to be in loading the plugins directory we use to load the optimizers. GoFSH 2.0.0 introduced a change to how the compiled JavaScript imports the optimizers directory, which causes an issue in FSH Online. (Note: If you want to see the issue, you can change the GoFSH dependency here to `2.0.0`, npm install and start the app, and try to run GoFSH on some FHIR resource.) The [GoFSH PR](https://github.com/FHIR/GoFSH/pull/225) avoids the issue. To test out that fix, you can check out the GoFSH branch, `npm pack`, and add the packed `.tgz` file as the dependency in this branch (for example: `"gofsh": "file:../gofsh/gofsh-2.0.0.tgz",` in `package.json`). When you start the app with this version of GoFSH, running GoFSH should work again, but there is a new warning in the console. I _think_ this warning is coming from the non-default case in the GoFSH code to load a provided directory of optimizers, but since FSH Online does not use that case, no negative effects are seen in FSH Online. However, if anyone has other ideas about how to resolve the issue coming from GoFSH v2.0.0 besides the PR there or ideas about how to resolve the warning that is coming up with that branch, I'm also very open to them.~

_**Update on the above paragraph**: This PR was updated to use the latest GoFSH release v2.0.1, which includes support for FSH Online._

Also, if you test this PR and notice issues that I missed that come from the warnings or other parts of the upgrade, let me know.

### CODE REVIEW

**Review by cmoesel — COMMENTED** [review]
I tested this and confirmed that it works as expected. While I did see two warnings when running this using the development server, I did not see them when running it as a static build.

I'll hold off on approving until the official GoFSH release has been integrated.

**Comment by jafeltra** [comment]
Whoops - forgot to push this up earlier, but I just added [d009e01](https://github.com/FSHSchool/FSHOnline/pull/136/commits/d009e012b57e8e0e4ed76349c090f1d32944e767), which should resolve one of the warnings (the one coming from GoFSH).

**Comment by jafeltra** [comment]
I've included the latest GoFSH and SUSHI versions and updated the description of this PR. It should be set for re-review.

**Review by cmoesel — CHANGES_REQUESTED** [review]
Sadly, I must report that I've found some issues... Since this is on SUSHI 3.1.0, I thought I'd try using the new `Context:` keyword, so I modified one of the examples to add context:
```
Extension: TherapySessionsCompleted
Id:        therapy-sessions-completed
Title:     "Therapy Sessions Completed"
Description: "The number of sessions of some therapy."
Context: Procedure
* value[x] only unsignedInt
```
When I convert to JSON, I get this error in the console:
```
error Error in parsing: ctx.start is undefined
  Line: 1 - 6
```

**Comment by jafeltra** [comment]
Well... that isn't good. I'll look into it.

**Comment by jafeltra** [comment]
> Oh! And since you're bringing in SUSHI 3.1.0, is it fair to ask for `Context:`, `Characteristics:`, and `CodeableReference(...)` to get the highlighting treatment too?

Very fair... and in fact I already added them locally!

**Comment by jafeltra** [comment]
Update for what is going on in this PR:

- [9f2844f](https://github.com/FSHSchool/FSHOnline/pull/136/commits/9f2844ffed3f5e57c5df33f9214c90db88e6af9f) adds the new highlighting for new keywords that are a part of the latest SUSHI release.
- [c0b5fd3](https://github.com/FSHSchool/FSHOnline/pull/136/commits/c0b5fd3d8029276b7172003abaad01284cb4938c) updates to SUSHI v3.2.0, which includes the update for ANTLR. Using this version resolves the error @cmoesel  mentioned above when using the `Context` keyword.

Right now, [c0b5fd3](https://github.com/FSHSchool/FSHOnline/pull/136/commits/c0b5fd3d8029276b7172003abaad01284cb4938c) seems to work in the app without issue. I have tested a few different FSH files and everything seems to be working fine, including the `Context` case that was an issue earlier. However, with the latest SUSHI update to 3.2.0, the tests do not seem to be configured correctly. I'm still working on how to resolve these, but I also wanted to push up the latest version to see if anyone notices any issues within the app that I missed in the meantime.

**Comment by jafeltra** [comment]
Aaaand... we're back!

I think this PR is finally up and running again! HUGE shoutout to @mint-thompson for our excellent teamwork to figure out the correct way to wrangle Jest.

The main update since the last comment on this PR is we figured out how to properly tell Jest which main antlr4 file to use within the testing environment. [7b29062](https://github.com/FSHSchool/FSHOnline/pull/136/commits/7b290627caada5265d1d71b86f4182b75345f6bd) adds a resolver for Jest to use antlr4's commonjs file when running tests in Jest. Some helpful links and documentation are included in the resolver file, which hopefully provide more details and explanation on the approach. If anything additional is needed, let me know.

[bedaaec](https://github.com/FSHSchool/FSHOnline/pull/136/commits/bedaaec3c0b58eef213c0e35d3cd2184f799398d) updates to the latest SUSHI and GoFSH versions.

I tested this with various FSH examples, including the one that used to break with the `Context` keyword, some other FSH 3.0 features, and some old examples I had hanging around. I feel pretty confident that the app itself works, and the tests are finally passing. Yay!

With that said... to the best of my knowledge, this PR finally does what it promised -- updates to a 3.x version of SUSHI and 2.x version of GoFSH. Let the re-reviews commence!

**Review by mint-thompson — APPROVED** [review]
Confirmed that the `Context` keyword works as expected when running locally. Hooray!

**Review by cmoesel — APPROVED** [review]
If it's good enough for @mint-thompson, it's good enough for me!
